app/webpacker/js/turbo.js

```diff
@@ -8,10 +8,23 @@ document.addEventListener("turbo:frame-missing", (event) => {
   event.preventDefault();
 
   // show error message instead
-  status = event.detail.response.status;
+  showError(event.detail.response?.status);
+});
+
+document.addEventListener("turbo:submit-end", (event) => {
+  if (!event.detail.success){
+    // show error message on failure
+    showError(event.detail.fetchResponse?.statusCode);
+    event.preventDefault();
+  }
+});
+
+function showError(status) {
   if(status == 401) {
     alert(I18n.t("errors.unauthorized.message"));
+  } else if(status === undefined) {
+    alert(I18n.t("errors.network_error.message"));
   } else {
```

Comment:
Looking at this again, I think we should not catch all 4xx (Client errors) with a generic message. They need specific handling and perhaps are handled differently in different parts of the app.

But we can catch all 5xx (server errors) and give a general message.

```suggestion
  } else if (status >= 500) {
```

(I'm not sure if status is a string or integer, but it will be sorted correctly either way)

How does that sound to you @rioug?

Reply:
I note that the existing `frame-missing` event handler would no longer cover the situation where an unexpected client error results in  no turbo-frame response. But Gaetan already tested that a 422 is handled correctly, so it should be fine 👍

Reply:
Sounds good :+1: